app/gui/view/component-browser/component-list-panel/grid/src/entry/style.rs

```diff
@@ -97,75 +176,72 @@ pub struct Style {
-        let color_intensities = color_intensities.clone_ref();
+        let bg = style_watch.get_color(panel_theme::background_color);
+        let colors = colors.clone_ref();
+        let color_anim = color::Animation::new(network);
 
-        let intensity = Animation::new(network);
         frp::extend! { network
             init <- source_();
 
-            text_intensity <- color_intensities.map(|c| c.text);
-            bg_intensity <- color_intensities.map(|c| c.background);
-            hover_hg_intensity <- color_intensities.map(|c| c.hover_highlight);
-            dimmed_intensity <- color_intensities.map(|c| c.dimmed);
-            icon_strong_intensity <- color_intensities.map(|c| c.icon_strong);
-            icon_weak_intensity <- color_intensities.map(|c| c.icon_weak);
-
-            one <- init.constant(1.0);
+            bg_intensity <- colors.map(|c| c.background_intensity);
+            bg <- all_with(&bg, &init, |col, ()| color::Lcha::from(col));
+            bg_and_main <- all(&bg, &color_anim.value);
             let is_dimmed = is_dimmed.clone_ref();
-            intensity.target <+ is_dimmed.switch(&one, &dimmed_intensity);
-            app_bg <- all_with(&app_bg, &init, |col, ()| color::Lcha::from(col));
-            app_bg_and_input <- all(&app_bg, color);
-            main <- app_bg_and_input.all_with(&intensity.value, mix);
-            app_bg_and_main <- all(&app_bg, &main);
-            background <- app_bg_and_main.all_with(&bg_intensity, mix).sampler();
-            hover_highlight <- app_bg_and_main.all_with(&hover_hg_intensity, mix).sampler();
-            text <- app_bg_and_main.all_with(&text_intensity, mix).sampler();
-            icon_weak <- app_bg_and_main.all_with(&icon_weak_intensity, mix).sampler();
-            icon_strong <- app_bg_and_main.all_with(&icon_strong_intensity, mix).sampler();
+            dimmed <- all_with3(&init, main_color, &colors,
+                |_, main, colors| colors.dimmed.resolve(main)
+            );
+            color_anim.target <+ switch(&is_dimmed, main_color, &dimmed);
+
+            // Component list panel does not support semi-transparent background. Instead, we mix
+            // the color of the component browser's background and the main color of the
+            // component group.
+            background <- all_with(&bg_and_main, &bg_intensity,
+                |(bg, main), intensity| color::mix(*bg, *main, *intensity)
+            ).sampler();
+            hover_highlight <- all_with(&color_anim.value, &colors,
+                |main, colors| colors.hover_highlight.resolve(main)
+            ).sampler();
+            text <- all_with(&color_anim.value, &colors,
+                |main, colors| colors.text.resolve(main)
+            ).sampler();
+            icon <- all_with(&color_anim.value, &colors,
+                |main, colors| colors.icon.resolve(main)
+            ).sampler();
```

Comment:
I cannot understand one thing: before we mixed all colors with application background. Now we rely just on alpha component.

But that means I cannot have background intensity 1 and color based on main color, can I? For example icon:
with bg_intensity 1 I get just the main color as entry background; then I display the icon being just a main color with some alpha. Then I think I will not see the icon, as main color with transparency should not be clearly visible over main color without.

Reply:
Yes, you're right, the icon would not be visible. Is this a serious concern? cc @wdanilo

Reply:
TBH, I don't understand the Adam's description above. I'd love to chat about it on a short call. But nevertheless, the correct way of blending icons (and other things) is with alpha. Icons should blend with background below them, which not always is a application bg. The node searcher color is different then app bg, component groups have colors, or node have colors. Correct blending then is by mixing the icon color with the background color (directly under icon). Setting opacity works exactly this way.